src/test/java/teammates/test/cases/webapi/ConfirmFeedbackSessionSubmissionActionTest.java

```diff
@@ -23,13 +50,256 @@ protected String getRequestMethod() {
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse1.getFeedbackSessionName());
+        verifyHttpParameterFailure(Const.ParamsNames.COURSE_ID, typicalCourse1.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse1.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true");
+
+        ______TS("Typical success case with student intent");
+
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        String[] studentParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.COURSE_ID, typicalCourse1.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse1.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+
+        ConfirmFeedbackSessionSubmissionAction a = getAction(studentParams);
+        JsonResult r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+
+        ConfirmationResponse cr = (ConfirmationResponse) r.getOutput();
+        assertEquals("Submission confirmed", cr.getMessage());
+
+        ______TS("Typical success case with student intent, not responded before");
+
+        loginAsStudent(student4InCourse1.getGoogleId());
+
+        for (FeedbackResponseAttributes res : allResponsesInSession1Course1) {
+            assertNotEquals(student4InCourse1.getEmail(), res.getGiver());
+        }
+
+        String[] studentNotRespondedParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.toString(),
+                Const.ParamsNames.COURSE_ID, typicalCourse1.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse1.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+
+        a = getAction(studentNotRespondedParams);
+        r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+
+        cr = (ConfirmationResponse) r.getOutput();
+        assertEquals("Submission confirmed", cr.getMessage());
+
+        ______TS("Typical success case with instructor intent");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+
+        String[] instructorParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.COURSE_ID, typicalCourse1.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session1InCourse1.getFeedbackSessionName(),
+                Const.ParamsNames.SEND_SUBMISSION_EMAIL, "true",
+        };
+
+        a = getAction(instructorParams);
+        r = getJsonResult(a);
+
+        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
+
+        cr = (ConfirmationResponse) r.getOutput();
+        assertEquals("Submission confirmed", cr.getMessage());
```

Comment:
verify email is sent?

Generally, you need to verify the side-effect of the action. e.g. email sent, response rate updated etc.